eslint.config.js

```diff
@@ -3,150 +3,279 @@ import { default as eslint } from "@eslint/js";
 import eslintPluginPrettierRecommended from "eslint-plugin-prettier/recommended";
```

Comment:
I have made this a source of truth for all of the linting and type checking disables that are happening throughout the code. this makes it so that we can actually see what is happening and stops us from hiding bugs from our tools.